src/flashops.c

```diff
@@ -99,16 +99,13 @@ int flashtobuf(uint8_t *buf, const char *file_path, ssize_t sz)
 	}
 }
 
-/* XXX: See if there is a way to only write data that changed? */
 int buftoflash(uint8_t *buf, const char *file_path, ssize_t sz)
 {
 	FILE *fd;
+	int ret;
 
-	/* XXX: Move this print to debug out */
-	/* XXX: Check error codes */
-	printf("Writing dataflash...\n");
 	fd = fopen(file_path, "wb");
-	fwrite(buf, sizeof(uint8_t), sz, fd);
+	ret = fwrite(buf, sizeof(uint8_t), sz, fd);
 	fclose(fd);
-	return 0;
+	return ret;
```

Comment:
If we care about the return value of `fwrite`, should we also care about the return value of `fopen` and `fclose`?

Reply:
The only error fclose() would return is if the fd was bad.
fopen() could be a valid return to care about, and this could be reworked to include that. However I figured making the upper layers care about the return would be more useful. Which leads to..
fwrite() will return the number of bytes written. This is what the upper layers would most care about IMO. If fopen() failed, then fwrite() will fail with 0 bytes written. If fclose() fails then there most likely would have been 0 bytes written still.

I'm open to further arguments about this. But to handle all of the cases I think it would be best to expand the buftoflash() function to do the documenting.